### DISCUSSION TITLE
TemporalValueRange exceptions

### DISCUSSION THREAD

**Discussion by mikepictor** [TimefoldAI/timefold-solver#1059] [Ideas]
I have multiple times wished I could add breaks in my TemporalValueRange. I specify a range of 9:00 to 17:00 for example for scheduling, but I want to skip 12:00 -12:30 when the lunch break is. 

It would be real nice if the TemporalValueRange can take a list of Pairs (I am using kolin, so the Pair entity, but whatever Java equivalent) representing start and end of the exceptions to the temporal range. 

As it is, I originally had a constraint devoted to catching anything in a break, though I recently threw out the TemporalValueRange in favour of a list of timestamps that exclude timesamps in the break ranges.

**Comment by triceo** [comment]
The `ValueRange`s haven't seen any love in a long time, and this is an interesting feature request. Thank you for the suggestion.

We see most people using lists, as that is the easiest thing to do. Since this can also be solved using a list, I don't see us working on this any time soon.

Pull requests are welcome though!
